libmscore/measure.cpp

```diff
@@ -3576,18 +3576,20 @@ void Measure::addSystemHeader(bool isFirstSystem)
                         }
                   else
                         keysig  = toKeySig(kSegment->element(track));
-                  if (!keysig) {
-                        //
-                        // create missing key signature
-                        //
-                        keysig = new KeySig(score());
-                        keysig->setTrack(track);
-                        keysig->setGenerated(true);
-                        keysig->setParent(kSegment);
-                        kSegment->add(keysig);
-                        }
-                  keysig->setKeySigEvent(keyIdx);
-                  keysig->layout();
+                  if (staff->staffType(tick())->genKeysig()) {
```

Comment:
So we were previously generating them here if the global staff property was set, and now we still will be for the global staff property but not if this option is set by staff type change?  That seems inconsistent.

Reply:
I'm not sure I understand what you mean by it being inconsistent. I made this change based on how time signatures handled this, and it made sense at first, but I definitely could be missing something. What would you propose instead?

Reply:
Right now, if you go to Staff Properties and turn off "Generate key signatures", it seems from this code that the code below still gets executed - correct?  But with you change, that same code does *not* get executed if instead you turn off the same property from a staff type change.  That's what seems executed.  Either that code should be executed in both cases or in neither.  I don't have a preference offhand, you should test it both ways and see what differences there might be.

Reply:
Oh actually I just realized there is no such thing as "Generate key signatures" in Staff Properties which is a bit interesting. The only way to change this property is through "Change staff type"

Reply:
Okay I just fixed this in the UI.  StaffType is what gets changed for both "Staff Properties" and "Change staff type". It's just in Staff Properties, it changes the properties from Fraction(0, 1) rather than a tick somewhere in the middle of the score. Because of this, the line of code I added before does in fact work for both cases.